tests/test_lag_2.py

```diff
@@ -40,244 +35,239 @@ def common_setup_teardown(duthost, ptfhost, testbed, conn_graph_facts):
+
+        # Figure out remote VM and interface info for the lag member and run minlink test
+        peer_device   = self.vm_neighbors[intf]['name']
+        neighbor_intf = self.vm_neighbors[intf]['port']
+        self.__verify_lag_minlink(self.nbrhosts[peer_device], lag_name, intf, neighbor_intf, po_interfaces, po_flap, deselect_time=95)
+
+    def run_lag_fallback_test(self, lag_name):
+        logging.info("Start checking lag fall back for: %s" % lag_name)
+
+        lag_facts           = self.__get_lag_facts()
+        intf, po_interfaces = self.__get_lag_intf_info(lag_facts, lag_name)
+        po_fallback         = lag_facts['lags'][lag_name]['po_config']['runner']['fallback']
+
+        # Figure out remote VM and interface info for the lag member and run lag fallback test
+        peer_device   = self.vm_neighbors[intf]['name']
+        neighbor_intf = self.vm_neighbors[intf]['port']
+        vm_host       = self.nbrhosts[peer_device]
+
+        try:
+            # Shut down neighbor interface
+            vm_host.shutdown(neighbor_intf)
+            time.sleep(120)
+
+            # Refresh lag facts
+            lag_facts = self.__get_lag_facts()
+
+            # Get teamshow result
+            teamshow_result = self.duthost.shell('teamshow')
+            logging.debug("Teamshow result: %s" % teamshow_result)
+
+            # Verify lag members
+            # 1. All other lag should keep selected state
+            # 2. Shutdown port should keep selected state if fallback enabled
+            # 3. Shutdown port should marded as deselected if fallback disabled
+            #  is marked deselected for the shutdown port and all other lag member interfaces are marked selected
+            for po_intf in po_interfaces.keys():
+                if po_intf != intf or po_fallback:
+                    assert lag_facts['lags'][lag_name]['po_stats']['ports'][po_intf]['runner']['selected']
+                else:
+                    assert not lag_facts['lags'][lag_name]['po_stats']['ports'][po_intf]['runner']['selected']
+
+            # The portchannel should marked Up/Down correctly according to po fallback setting
+            if po_fallback:
+                assert lag_facts['lags'][lag_name]['po_intf_stat'] == 'Up'
+            else:
+                assert lag_facts['lags'][lag_name]['po_intf_stat'] == 'Down'
 
-def test_lag_2(common_setup_teardown, nbrhosts):
-    duthost, ptfhost, vm_neighbors, mg_facts, lag_facts, fanout_neighbors = common_setup_teardown
+        finally:
+            # Bring up neighbor interface
+            vm_host.no_shutdown(neighbor_intf)
+            time.sleep(30)
+
+            # Refresh lag facts
+            lag_facts = self.__get_lag_facts()
+
+            # Verify all interfaces in port_channel are marked up
+            for po_intf in po_interfaces.keys():
+                assert lag_facts['lags'][lag_name]['po_stats']['ports'][po_intf]['link']['up'] == True
```

Comment:
I suggest remove these asserts (and the one below) from finally block. This block is to restore whatever test did to leave a clean environment for next test case.

Will asserting here cause noise in debugging when test fails?

Reply:
The assert here was from origin ansible playbook test.
I think I can use wait_until to check if the teatbed env are cleaned, and raise exception if clean up fail instead of use assert.